### ISSUE TITLE
Appium Support

### ISSUE THREAD

**Issue by sumanthmattapali** [pazone/ashot#112]
Hi Guys,

This is more enhancement than an issue. I am just wondering whether Ashot supports android and ios apps?

**Comment by rickynugraha** [comment]
It works with AndroidDriver, probably with some limitations. It works fine so far on my apps.

IosDriver is unfortunately not working. It will ask you to change to WebDriver instead.

**Comment by klementynka** [comment]
This question is very interesting for me too :)

**Comment by sachingainewar** [comment]
Yes i am also interested to know.. i tried this but end up so many errors

**Comment by Kiera11** [comment]
Hi,

Any update on this does Ashot supports android and ios apps?

Thanks,

**Comment by mivola** [comment]
Hi guys,

anyone tried to use it for native apps on Android/iOS (i.e. using MobileElement instead of WebElement) yet?

Thanks
Michael

**Comment by vergilfrans** [comment]
I trying to use ashot for mobile applications. I used Mobiledriver instead of webdriver but so far, no luck.  Is it working for anyone?

**Comment by jayantsarda** [comment]
ShootingStrategies class of ashot has implementations for various iOS and Android(nonRetina) did you gave a try to those ?

**Comment by vergilfrans** [comment]
> ShootingStrategies class of ashot has implementations for various iOS and Android(nonRetina) did you gave a try to those ?

Hi Jayantsarda, I was not aware of it.  I will try shootingstrategies...Thank you very much for sharing this info.
I have to take screenshot of the webelements(icons, images, logos) and compare them with the actual images. Guess, I am trying the right set of tools - katalon studio, appium with ashot. Kindly guide me if you have better solutions.

**Comment by jayantsarda** [comment]
As far as I have understood the Ashot lib so far :
1. It needs a shooting strategy to adjust how and where you want to take screenshot and takes care of your aspect ratio , retina display etc
2. It captures the screen shot for the whole screen (under the hood it used webdriver screenshot capture utility or using java script.  You can implement your own screen capture utility 
3. It tries to find the coordinates of element you pass (again can uses webdriver get coordinates for this )
4. than it crops out the section of webelement from the screen image captured  in step 2.
I have not tried with appium driver but i am pretty sure there should be equivalent methods to achieve these steps.

**Comment by vergilfrans** [comment]
Hi,

Thank you very much for the step-by-step explanation on how Ashot works. I
will explore further based on your guidance.

**Comment by jaskaran0001** [comment]
Is there any update on this:
Can we use Ashot to capture full-page scrolling screenshot for Native apps?
Is there any other library that supports this?

**Comment by itkhanz** [comment]
I came across this library and have same question. Does this library support screenshot testing for native mobile apps (iOS and Android)? I would like to know if it supports capturing screenshots of WebElement on mobile apps and whether we can perform the utility actions provided by this library also for mobile automation?

**Comment by valfirst** [comment]
@itkhanz yes, it is

**Comment by itkhanz** [comment]
> @itkhanz yes, it is

Thank you. README of this project does not indicate that it also supports mobile applications so i was unsure.

I am looking for ways to do screenshote testing. Have you used this tool for image comparison? I checked that it provides some helper methods for image comparison which may be helpful for my needs
https://github.com/pazone/ashot/?tab=readme-ov-file#screenshot-comparison

Appium also provides some utility methods to do visual testing with images plugin:
https://github.com/appium/appium/blob/master/packages/images-plugin/docs/image-comparison.md

If you have tried both or any, could you share your experience which one is more suitable for image comparison?

**Comment by valfirst** [comment]
> Have you used this tool for image comparison?

yes

> If you have tried both

no

> could you share your experience which one is more suitable for image comparison?

I have no idea
